<?xml version='1.0' encoding='utf-8'?>
<launch>

    <!--node pkg="palletizing" type="Palletizing_NODE" name="TD_Node">
    </node-->

    <!-- USB相机及AR码节点  -->
    <include file="$(find usb_cam)/launch/ar_track_alvar_test.launch"/>

    <!--  socket图传  -->
    <node pkg="td_palletizing" type="client_ros.py" name="IMG_Node">
        <param name="host_ip" value="192.168.79.198" />
        <param name="host_port" value="9999" />
    </node>
    
    <!--  塔吊抓取程序  -->
    <node pkg="td_palletizing" type="TA_Mission_two.py" name="Mission_Node">
        <param name="host_ip" value="192.168.79.198" />
        <param name="host_port" value="50002" />

        <!--视�?�识�?模式, 1代表AGV上有�?(识别放置,默�??6号码),0表示无码(盲放)-->
        <param name="vision_mode" value="0" />

	    <!--抓取集�?��??1识别�?,单位mm-->
	    <param name="grasp_x1" value="18.0"/>
	    <param name="grasp_y1" value="0.0"/>
	    <param name="grasp_z1" value="0.0"/>

	    <!--侧边放置集�?��??1位置�?,单位mm-->
	    <param name="push_x1" value="206"/>
	    <param name="push_y1" value="378"/>
	    <param name="push_z1" value="210.0"/>
	
	    <!--货车默�?�集装�??1放置�?-->
	    <param name="real_push_x1" value="15.0"/>
	    <param name="real_push_y1" value="-45.0"/>
	    <param name="real_push_z1" value="380.0"/>

	    <!--抓取集�?��??2识别�?,单位mm-->
	    <param name="grasp_x2" value="35.0"/>
	    <param name="grasp_y2" value="0.0"/>
	    <param name="grasp_z2" value="0.0"/>

	    <!--侧边放置集�?��??2位置�?,单位mm-->
	    <param name="push_x2" value="-94"/>
	    <param name="push_y2" value="378.0"/>
	    <param name="push_z2" value="210.0"/>
	
	    <!--货车默�?�集装�??2放置�?-->
	    <param name="real_push_x2" value="15.0"/>
	    <param name="real_push_y2" value="-45.0"/>
	    <param name="real_push_z2" value="380.0"/>

    </node>
    
    <!-- 塔吊识别程序 -->
    <node pkg="td_palletizing" type="TA_Vision_two.py" name="Vision_Node">
        <param name="host_ip" value="192.168.79.198" />
        <param name="host_port" value="50002" />
        <!--视�?�识�?模式, 1代表AGV上有�?(识别放置,默�??6号码),0表示无码(盲放)-->
        <param name="vision_mode" value="0" />

	    <!--视�?�补�?,单位mm-->
        <param name="x1_dis" value="0" />
        <param name="y1_dis" value="0" />
        <param name="z1_dis" value="0" />

        <param name="x2_dis" value="0" />
        <param name="y2_dis" value="0" />
        <param name="z2_dis" value="0" />

    </node>

</launch>
